common/ledger/blockledger/fileledger/factory_test.go

```diff
@@ -36,16 +41,29 @@ func TestBlockStoreProviderErrors(t *testing.T) {
 	})
 
 	t.Run("open", func(t *testing.T) {
-		mockBlockStore.OpenReturns(nil, errors.New("woogie"))
+		mockBlockStoreProvider.OpenReturns(nil, errors.New("woogie"))
 		_, err := f.GetOrCreate("foo")
 		require.EqualError(t, err, "woogie")
 		require.Empty(t, f.ledgers, "Expected no new ledger is created")
 	})
 
 	t.Run("remove", func(t *testing.T) {
-		mockBlockStore.DropReturns(errors.New("oogie"))
-		err := f.Remove("foo")
-		require.EqualError(t, err, "oogie")
+		t.Run("ledger doesn't exist", func(t *testing.T) {
+			err := f.Remove("foo")
+			require.NoError(t, err)
+			require.Equal(t, 1, mockBlockStoreProvider.DropCallCount())
+		})
+
+		t.Run("dropping the blockstore fails", func(t *testing.T) {
+			mockBlockStore := &mock.FileLedgerBlockStore{}
+			f.ledgers["foo"] = &FileLedger{blockStore: mockBlockStore}
+			mockBlockStoreProvider.DropReturns(errors.New("oogie"))
+
+			err := f.Remove("foo")
+			require.EqualError(t, err, "oogie")
+			require.Equal(t, 1, mockBlockStore.ShutdownCallCount())
+			require.Equal(t, 2, mockBlockStoreProvider.DropCallCount())
```

Comment:
It's cleaner to write sub-test that makes them independent of each other. If you run this sub-test independently from outside `go test -run TestBlockStoreProviderErrors/remove/dropping_the_blockstore_fails`, Or if someone changes the order of subtests,  it will fail.

My suggestion would be to have a clean setup for each subtest... Here are a couple patterns that we have been following for this - [1](https://github.com/hyperledger/fabric/blob/f79abaa110fc6d6d6089b183ce62333afb5a6368/common/ledger/util/leveldbhelper/leveldb_provider_test.go#L387) or [2](https://github.com/hyperledger/fabric/blob/f79abaa110fc6d6d6089b183ce62333afb5a6368/core/ledger/pvtdatastorage/snapshot_data_importer_test.go#L57)